test/SpokePoolClient.ValidateFill.ts

```diff
@@ -0,0 +1,591 @@
+      5,
+      spokePool1DeploymentBlock,
+      spokePoolClient1.latestBlockNumber,
+      10
+    );
+
+    expect(searchRange3.high).to.be.greaterThanOrEqual(depositEvents[2].blockNumber);
+    expect(searchRange3.low).to.be.lessThanOrEqual(depositEvents[2].blockNumber);
+    expect(searchRange4.high).to.be.greaterThanOrEqual(depositEvents[3].blockNumber);
+    expect(searchRange4.low).to.be.lessThanOrEqual(depositEvents[3].blockNumber);
+    expect(searchRange5.high).to.be.greaterThanOrEqual(depositEvents[4].blockNumber);
+    expect(searchRange5.low).to.be.lessThanOrEqual(depositEvents[4].blockNumber);
+  });
+
+  it("Fuzz: get search bounds for deposit ID", async function () {
+    const fuzzClient = new MockSpokePoolClient(
+      createSpyLogger().spyLogger,
+      spokePool_2,
+      destinationChainId,
+      spokePool2DeploymentBlock
+    );
+
+    const initLow = 0;
+    const initHigh = 1000000;
+    const depositIds = Array(initHigh - initLow + 1).fill(0);
+
+    const testIterations = 1000;
+
+    // Randomize deposit ID's between initLow and initHigh. ID's should only increase
+    // and will do so 50% of the time. The other 50% of the time they will stay the same.
+    // The increment will be between 0 and 10, to simulate sending multiple deposits in the same
+    // block.
+    for (let i = 1; i < depositIds.length; i++) {
+      const increment = Math.max(0, Math.floor((Math.random() - 0.5) * 10));
+      depositIds[i] = depositIds[i - 1] + increment;
+    }
+    fuzzClient.setDepositIds(depositIds);
+
+    for (let i = 0; i < testIterations; i++) {
+      // Randomize target between highest and lowest values in deposit IDs.
+      const target = Math.floor(Math.random() * (depositIds[depositIds.length - 1] - initLow)) + initLow;
+
+      // Randomize max # of searches.
+      const maxSearches = Math.floor(Math.random() * 19) + 1;
+      const results = await fuzzClient._getBlockRangeForDepositId(target, initLow, initHigh, maxSearches);
+
+      // The correct block is the first block to increment beyond the target. Check that the range
+      // contains the correct block where deposit ID incremented to the target + 1.
+      const correctBlock = depositIds.findIndex((depositId) => depositId > target);
+      expect(depositIds[correctBlock] > target).to.be.true;
+      expect(results.low <= correctBlock).to.be.true;
+      expect(results.high >= correctBlock).to.be.true;
+
+      // Check other invariants.
+      expect(results.low >= initLow).to.be.true;
+      expect(results.high <= initHigh).to.be.true;
+
+      // The block range should return a range containing:
+      // - a low block with deposit count <= target
+      // - a high block with deposit count > target.
+      expect(depositIds[results.low] <= target).to.be.true;
+      expect(depositIds[results.high] > target).to.be.true;
+    }
+  });
+
+  // TODO: Fix these tests. The reason is that we don't have the historical
```

Comment:
Did you comment this out or was this already commented?

Reply:
This **was** a change that I've added. I've recorded it in ACX-1563 so that it's not forgotten.